lib/opal/builder_processors.rb

```diff
@@ -27,31 +28,31 @@ def self.handles(*extensions)
         nil
       end
 
-      def self.extensions
-        @extensions
+      class << self
+        attr_reader :extensions
       end
```

Comment:
I agree but overall it may be consistent because sometimes we are using:

```ruby
class << self
  def foo
  end
end
```

And sometimes:

```ruby
def self.foo
end
```

Reply:
Method that returns an ivar IS `attr_reader`. Could you please explain why do you like previous version more?

Reply:
sure, `class << self` and `attr_reader` are harder to parse when reading the code

the meaning is more immediate in the first form, without the new scope plus metaprogramming

As I think it's already clear, I'm not arguing for a rule against `class << self` or `attr_reader` but rather about setting a rule that can worsen the readability and force us to workarounds just to appease the machine (e.g. `# else noop`).